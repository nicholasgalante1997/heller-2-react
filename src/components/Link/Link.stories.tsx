import React from 'react';
import { type Meta, type StoryObj } from '@storybook/react';

import { Link } from './Link';

const meta: Meta<typeof Link> = {
  title: 'Link',
  component: Link,
  decorators: [],
  args: {
    children: 'For more information, visit our messageboard.',
    href: '#'
  }
};

export default meta;

export const Main: StoryObj<typeof Link> = {
  render: (args) => <Link {...args} />
};
