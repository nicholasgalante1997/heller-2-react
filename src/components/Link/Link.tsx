import React from 'react';
import classNames from 'classnames';

import LinkProps from './Link.types';

const linkClassName = 'h-link';

function LinkComponent({ children, className, ...rest }: LinkProps) {
    return (
        <a {...rest} className={classNames({ [linkClassName]: true, className })}>
            {children}
        </a>
    );
}

export const Link = React.memo(LinkComponent as React.FC<LinkProps>);
Link.displayName = 'HlrLink';