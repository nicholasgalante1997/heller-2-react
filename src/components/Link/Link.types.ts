import { type HTMLProps } from 'react';
type LinkProps = HTMLProps<HTMLAnchorElement>;
export default LinkProps;