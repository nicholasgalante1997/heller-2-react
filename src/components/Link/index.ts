export { Link } from './Link';
export { default as LinkProps } from './Link.types';