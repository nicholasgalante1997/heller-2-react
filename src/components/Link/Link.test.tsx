import React from 'react';
import { render } from '@testing-library/react';
import { Link } from './Link';

describe('Link', () => {
  test('Snapshot', () => {
    const { container } = render(<Link>More Information</Link>);
    expect(container).toMatchSnapshot();
  });
});
