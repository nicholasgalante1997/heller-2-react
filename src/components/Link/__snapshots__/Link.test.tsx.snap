// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link Snapshot 1`] = `
<div>
  <a
    class="h-link"
  >
    More Information
  </a>
</div>
`;
